--- a/src/storage/browser/quota/quota_settings.cc      2018-08-30 15:11:45.052446318 +0200
+++ b/src/storage/browser/quota/quota_settings.cc      2019-07-29 17:29:49.765826330 +0200
@@ -34,7 +34,7 @@
   const int64_t kMBytes = 1024 * 1024;
   const int kRandomizedPercentage = 10;
 
-  if (is_incognito) {
+  /*if (is_incognito) {
     // The incognito pool size is a fraction of the amount of system memory,
     // and the amount is capped to a hard limit.
     const double kIncognitoPoolSizeRatio = 0.1;  // 10%
@@ -49,7 +49,7 @@
     settings.session_only_per_host_quota = settings.per_host_quota;
     settings.refresh_interval = base::TimeDelta::Max();
     return settings;
-  }
+  }*/
 
 // The fraction of the device's storage the browser is willing to
 // use for temporary storage.
