diff -crB phpbms/install/createtables.sql phpbms_timestamp/install/createtables.sql
*** phpbms/install/createtables.sql	2009-12-31 20:36:45.000000000 +0000
--- phpbms_timestamp/install/createtables.sql	2011-11-03 12:47:29.000000000 +0000
***************
*** 36,42 ****
    createdby int(11) NOT NULL default '0',
    modifiedby int(11) default '0',
    creationdate datetime NOT NULL default '0000-00-00 00:00:00',
!   modifieddate timestamp(14) NOT NULL,
    `roleid` VARCHAR(64),
    PRIMARY KEY  (id),
    UNIQUE KEY (`uuid`)
--- 36,42 ----
    createdby int(11) NOT NULL default '0',
    modifiedby int(11) default '0',
    creationdate datetime NOT NULL default '0000-00-00 00:00:00',
!   modifieddate timestamp NOT NULL,
    `roleid` VARCHAR(64),
    PRIMARY KEY  (id),
    UNIQUE KEY (`uuid`)
***************
*** 116,122 ****
    createdby int(11) NOT NULL default '0',
    creationdate datetime NOT NULL default '0000-00-00 00:00:00',
    modifiedby int(11) default '0',
!   modifieddate timestamp(14) NOT NULL,
    inherint tinyint(1) NOT NULL default '0',
    PRIMARY KEY  (id),
    UNIQUE KEY (`uuid`),
--- 116,122 ----
    createdby int(11) NOT NULL default '0',
    creationdate datetime NOT NULL default '0000-00-00 00:00:00',
    modifiedby int(11) default '0',
!   modifieddate timestamp NOT NULL,
    inherint tinyint(1) NOT NULL default '0',
    PRIMARY KEY  (id),
    UNIQUE KEY (`uuid`),
***************
*** 267,273 ****
    createdby int(11) NOT NULL default '0',
    modifiedby int(11) default '0',
    lastlogin datetime default NULL,
!   modifieddate timestamp(14) NOT NULL,
    email varchar(128) default '',
    phone varchar(32) default '',
    department varchar(128) default '',
--- 267,273 ----
    createdby int(11) NOT NULL default '0',
    modifiedby int(11) default '0',
    lastlogin datetime default NULL,
!   modifieddate timestamp NOT NULL,
    email varchar(128) default '',
    phone varchar(32) default '',
    department varchar(128) default '',
***************
*** 318,324 ****
    `createdby` int(11) default '0',
    `creationdate` datetime default '0000-00-00 00:00:00',
    `modifiedby` int(11) default '0',
!   `modifieddate` timestamp(14) NOT NULL,
    `roleid` VARCHAR(64),
    `custom1` DOUBLE,
    `custom2` DOUBLE,
--- 318,324 ----
    `createdby` int(11) default '0',
    `creationdate` datetime default '0000-00-00 00:00:00',
    `modifiedby` int(11) default '0',
!   `modifieddate` timestamp NOT NULL,
    `roleid` VARCHAR(64),
    `custom1` DOUBLE,
    `custom2` DOUBLE,
***************
*** 340,346 ****
    `createdby` int(11) default '0',
    `creationdate` datetime default '0000-00-00 00:00:00',
    `modifiedby` int(11) default '0',
!   `modifieddate` timestamp(14) NOT NULL,
    PRIMARY KEY  (`id`),
    KEY `therecord` (`recordid`),
    KEY `thetable` (`tabledefid`),
--- 340,346 ----
    `createdby` int(11) default '0',
    `creationdate` datetime default '0000-00-00 00:00:00',
    `modifiedby` int(11) default '0',
!   `modifieddate` timestamp NOT NULL,
    PRIMARY KEY  (`id`),
    KEY `therecord` (`recordid`),
    KEY `thetable` (`tabledefid`),
diff -crB phpbms/modules/bms/install/createtables.sql phpbms_timestamp/modules/bms/install/createtables.sql
*** phpbms/modules/bms/install/createtables.sql	2010-01-06 00:06:44.000000000 +0000
--- phpbms_timestamp/modules/bms/install/createtables.sql	2011-11-03 17:26:20.000000000 +0000
***************
*** 66,72 ****
    description text,
    createdby int(11) default NULL,
    modifiedby int(11) default NULL,
!   modifieddate timestamp(14) NOT NULL,
    creationdate datetime default NULL,
    `custom1` DOUBLE,
    `custom2` DOUBLE,
--- 66,72 ----
    description text,
    createdby int(11) default NULL,
    modifiedby int(11) default NULL,
!   modifieddate timestamp NOT NULL,
    creationdate datetime default NULL,
    `custom1` DOUBLE,
    `custom2` DOUBLE,
***************
*** 171,177 ****
    createdby int(11) NOT NULL default '0',
    creationdate datetime NOT NULL default '0000-00-00 00:00:00',
    modifiedby int(11) default NULL,
!   modifieddate timestamp(14) NOT NULL,
    `custom1` DOUBLE,
    `custom2` DOUBLE,
    `custom3` DATETIME,
--- 171,177 ----
    createdby int(11) NOT NULL default '0',
    creationdate datetime NOT NULL default '0000-00-00 00:00:00',
    modifiedby int(11) default NULL,
!   modifieddate timestamp NOT NULL,
    `custom1` DOUBLE,
    `custom2` DOUBLE,
    `custom3` DATETIME,
***************
*** 216,222 ****
    `createdby` int(11) NOT NULL default 0,
    `creationdate` datetime NOT NULL default '0000-00-00 00:00:00',
    `modifiedby` int(11) default NULL,
!   `modifieddate` timestamp(14) NOT NULL,
    `custom1` DOUBLE,
    `custom2` DOUBLE,
    `custom3` DATETIME,
--- 216,222 ----
    `createdby` int(11) NOT NULL default 0,
    `creationdate` datetime NOT NULL default '0000-00-00 00:00:00',
    `modifiedby` int(11) default NULL,
!   `modifieddate` timestamp NOT NULL,
    `custom1` DOUBLE,
    `custom2` DOUBLE,
    `custom3` DATETIME,
***************
*** 240,246 ****
    isprepackaged tinyint(4) NOT NULL default '0',
    packagesperitem double default NULL,
    modifiedby int(11) default NULL,
!   modifieddate timestamp(14) NOT NULL,
    partname varchar(128) default NULL,
    partnumber varchar(32) NOT NULL default '',
    status varchar(32) NOT NULL default 'In Stock',
--- 240,246 ----
    isprepackaged tinyint(4) NOT NULL default '0',
    packagesperitem double default NULL,
    modifiedby int(11) default NULL,
!   modifieddate timestamp NOT NULL,
    partname varchar(128) default NULL,
    partnumber varchar(32) NOT NULL default '',
    status varchar(32) NOT NULL default 'In Stock',
***************
*** 283,289 ****
    createdby int(11) NOT NULL default '0',
    creationdate datetime NOT NULL default '0000-00-00 00:00:00',
    modifiedby int(11) default NULL,
!   modifieddate timestamp(14) NOT NULL,
    `custom1` DOUBLE,
    `custom2` DOUBLE,
    `custom3` DATETIME,
--- 283,289 ----
    createdby int(11) NOT NULL default '0',
    creationdate datetime NOT NULL default '0000-00-00 00:00:00',
    modifiedby int(11) default NULL,
!   modifieddate timestamp NOT NULL,
    `custom1` DOUBLE,
    `custom2` DOUBLE,
    `custom3` DATETIME,
