From 09175804c52e4d80b073a35bbe03edc1e51206e8 Mon Sep 17 00:00:00 2001
From: Xiaowu Zhang <xiaowu.zhang@nexedi.com>
Date: Mon, 25 Aug 2025 14:48:38 +0200
Subject: [PATCH] fix unicode in login/password

---
 src/AccessControl/userfolder.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/AccessControl/userfolder.py b/src/AccessControl/userfolder.py
index 3115928..363ce54 100644
--- a/src/AccessControl/userfolder.py
+++ b/src/AccessControl/userfolder.py
@@ -123,7 +123,7 @@ class BasicUserFolder(Implicit, Persistent, RoleManager):
                 name, password = decodebytes(auth.split(b' ')[-1]) \
                     .decode().split(':', 1)
             except:
-                raise BadRequest('Invalid authentication token')
+                return None, None
             return name, password
         else:
             return None, None
-- 
2.48.0

