--- glib/gstrfuncs.c~	2011-09-10 22:57:34.802250000 +0800
+++ glib/gstrfuncs.c	2011-09-10 22:57:52.630375000 +0800
@@ -1424,7 +1424,7 @@ g_strsignal (gint signum)
   const char *msg_locale;
 
 #if defined(G_OS_BEOS) || defined(G_WITH_CYGWIN)
-extern const char *strsignal(int);
+extern char *strsignal(int);
 #else
   /* this is declared differently (const) in string.h on BeOS */
   extern char *strsignal (int sig);
