Origin: https://github.com/arskom/spyne/pull/704
From: Burak Arslan <burak.arslan@arskom.com.tr>
Date: Mon, 28 Nov 2016 11:02:24 +0300
Subject: re-add customizations to the bundled six module

---
 spyne/util/six.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/spyne/util/six.py b/spyne/util/six.py
index d4fe9849f..84871a9c7 100644
--- a/spyne/util/six.py
+++ b/spyne/util/six.py
@@ -363,6 +363,8 @@ class Module_six_moves_urllib_parse(_LazyModule):
     MovedAttribute("splittag", "urllib", "urllib.parse"),
     MovedAttribute("splituser", "urllib", "urllib.parse"),
     MovedAttribute("splitvalue", "urllib", "urllib.parse"),
+    MovedAttribute("splittype", "urllib", "urllib.parse"),
+    MovedAttribute("splithost", "urllib", "urllib.parse"),
     MovedAttribute("uses_fragment", "urlparse", "urllib.parse"),
     MovedAttribute("uses_netloc", "urlparse", "urllib.parse"),
     MovedAttribute("uses_params", "urlparse", "urllib.parse"),
@@ -531,6 +533,7 @@ def remove_move(name):
     _func_code = "__code__"
     _func_defaults = "__defaults__"
     _func_globals = "__globals__"
+    _func_name = "__name__"
 else:
     _meth_func = "im_func"
     _meth_self = "im_self"
@@ -539,6 +542,7 @@ def remove_move(name):
     _func_code = "func_code"
     _func_defaults = "func_defaults"
     _func_globals = "func_globals"
+    _func_name = "func_name"
 
 
 try:
@@ -592,6 +596,7 @@ def next(self):
 get_function_code = operator.attrgetter(_func_code)
 get_function_defaults = operator.attrgetter(_func_defaults)
 get_function_globals = operator.attrgetter(_func_globals)
+get_function_name = operator.attrgetter(_func_name)
 
 
 if PY3:
