From 77f86b50f097dcf364e0d140e45593bf001d46bc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=A9rome=20Perrin?= <jerome@nexedi.com>
Date: Fri, 1 Mar 2024 09:49:17 +0900
Subject: [PATCH] set metadata in setup.py for compatibility with old slapos
 buildout

---
 setup.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/setup.py b/setup.py
index 1bf0bcff5..a93fe7b22 100755
--- a/setup.py
+++ b/setup.py
@@ -987,6 +987,11 @@ ext_modules = [
 
 try:
     setup(
+        name='pillow',
+        version='10.2.0',
+        packages=["PIL"],
+        include_package_data=True,
+        package_dir={"": "src"},
         cmdclass={"build_ext": pil_build_ext},
         ext_modules=ext_modules,
         zip_safe=not (debug_build() or PLATFORM_MINGW),
-- 
2.42.0

