diff --git a/chuser.c b/chuser.c
index 0d79dbc..ccc3acc 100644
--- a/chuser.c
+++ b/chuser.c
@@ -11,6 +11,7 @@
 
 Prototype int ChangeUser(const char *user, char *dochdir);
 
+#if 0
 int
 ChangeUser(const char *user, char *dochdir)
 {
@@ -57,5 +58,7 @@ ChangeUser(const char *user, char *dochdir)
 		}
 	}
 	return(pas->pw_uid);
+	return getpwnam(user);
 }
+#endif
 
diff --git a/crontab.c b/crontab.c
index b29e0d2..06df70f 100644
--- a/crontab.c
+++ b/crontab.c
@@ -88,7 +88,7 @@ main(int ac, char **av)
 				break;
 			case 'c':
 				/* getopt guarantees optarg != 0 here */
-				if (*optarg != 0 && getuid() == geteuid()) {
+				if (*optarg != 0) {
 					CDir = optarg;
 				} else {
 					printlogf(0, "-c option: superuser only\n");
@@ -318,9 +318,6 @@ GetReplaceStream(const char *user, const char *file)
 
 	close(filedes[0]);
 
-	if (ChangeUser(user, NULL) < 0)
-		exit(0);
-
 	fd = open(file, O_RDONLY);
 	if (fd < 0) {
 		printlogf(0, "unable to open %s: %s", file, strerror(errno));
@@ -346,8 +343,6 @@ EditFile(const char *user, const char *file)
 		const char *ptr;
 		char visual[SMALL_BUFFER];
 
-		if (ChangeUser(user, TMPDIR) < 0)
-			exit(0);
 		if ((ptr = getenv("EDITOR")) == NULL || strlen(ptr) >= sizeof(visual))
 			if ((ptr = getenv("VISUAL")) == NULL || strlen(ptr) >= sizeof(visual))
 				ptr = PATH_VI;
diff --git a/job.c b/job.c
index b5e02de..79650c2 100644
--- a/job.c
+++ b/job.c
@@ -63,14 +63,6 @@ RunJob(CronFile *file, CronLine *line)
 		 * Change running state to the user in question
 		 */
 
-		if (ChangeUser(file->cf_UserName, TempDir) < 0) {
-			printlogf(LOG_ERR, "unable to ChangeUser (user %s %s)\n",
-					file->cf_UserName,
-					line->cl_Description
-					);
-			exit(0);
-		}
-
 		/* from this point we are unpriviledged */
 
 		if (DebugOpt)
@@ -296,14 +288,6 @@ EndJob(CronFile *file, CronLine *line, int exit_status)
 		 * by the mailing and we already verified the mail file.
 		 */
 
-		if (ChangeUser(file->cf_UserName, TempDir) < 0) {
-			printlogf(LOG_ERR, "unable to ChangeUser to send mail (user %s %s)\n",
-					file->cf_UserName,
-					line->cl_Description
-					);
-			exit(0);
-		}
-
 		/* from this point we are unpriviledged */
 
 		/*
diff --git a/Makefile b/Makefile
index 547febc..13829e6 100644
--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,6 @@ VERSION = 4.5
 
 # these variables can be configured by e.g. `make SCRONTABS=/different/path`
 PREFIX = /usr/local
-CRONTAB_GROUP = wheel
 SCRONTABS = /etc/cron.d
 CRONTABS = /var/spool/cron/crontabs
 CRONSTAMPS = /var/spool/cron/cronstamps
@@ -20,10 +19,10 @@ MANDIR = $(PREFIX)/share/man
 
 
 SHELL = /bin/sh
-INSTALL = install -o root
+INSTALL = install
 INSTALL_PROGRAM = $(INSTALL) -D
-INSTALL_DATA = $(INSTALL) -D -m0644 -g root
-INSTALL_DIR = $(INSTALL) -d -m0755 -g root
+INSTALL_DATA = $(INSTALL) -D -m0644
+INSTALL_DIR = $(INSTALL) -d -m0755
 CFLAGS ?= -O2
 CFLAGS += -Wall -Wextra -Wstrict-prototypes -Wno-missing-field-initializers -Wfloat-equal -fstack-protector-all -Wformat-security -Wformat=2 -fPIE 
 CFLAGS += -Wl,-z,nodump -Wl,-z,noexecstack -Wl,-z,noexecheap -Wl,-z,relro -Wl,-z,now -Wl,-z,nodlopen -Wl,-z,-pie
@@ -47,7 +46,6 @@ all: $(PROTOS) crond crontab ;
 	echo "SBINDIR = $(SBINDIR)" >> config
 	echo "BINDIR = $(BINDIR)" >> config
 	echo "MANDIR = $(MANDIR)" >> config
-	echo "CRONTAB_GROUP = $(CRONTAB_GROUP)" >> config
 	echo "SCRONTABS = $(SCRONTABS)" >> config
 	echo "CRONTABS = $(CRONTABS)" >> config
 	echo "CRONSTAMPS = $(CRONSTAMPS)" >> config
@@ -65,13 +63,10 @@ crontab: $(TABOBJS)
 	$(CC) $(CFLAGS) $(CPPFLAGS) -c $(DEFS) $< -o $@
 
 install:
-	$(INSTALL_PROGRAM) -m0700 -g root crond $(DESTDIR)$(SBINDIR)/crond
-	$(INSTALL_PROGRAM) -m4750 -g $(CRONTAB_GROUP) crontab $(DESTDIR)$(BINDIR)/crontab
+	$(INSTALL_PROGRAM) -m0755 crond $(DESTDIR)$(SBINDIR)/crond
+	$(INSTALL_PROGRAM) -m4750 crontab $(DESTDIR)$(BINDIR)/crontab
 	$(INSTALL_DATA) crontab.1 $(DESTDIR)$(MANDIR)/man1/crontab.1
 	$(INSTALL_DATA) crond.8 $(DESTDIR)$(MANDIR)/man8/crond.8
-	$(INSTALL_DIR) $(DESTDIR)$(SCRONTABS)
-	$(INSTALL_DIR) $(DESTDIR)$(CRONTABS)
-	$(INSTALL_DIR) $(DESTDIR)$(CRONSTAMPS)
 
 clean: force
 	rm -f *.o $(PROTOS)
