--- configure.ac.orig	2011-12-06 12:54:26.812408976 +0900
+++ configure.ac	2011-12-06 13:17:39.060465045 +0900
@@ -70,9 +70,10 @@
           m4_pattern_allow(PKG_CONFIG_LIBDIR)])
 
 AC_DEFUN([CHECK_CFLAG], [
+  cflag=$(echo "$1" | sed -e 's,^-Wno-,-W,')
   AC_MSG_CHECKING([if gcc supports $1])
   old_CFLAGS=$CFLAGS
-  CFLAGS="$CFLAGS $1 -Werror"
+  CFLAGS="$CFLAGS $cflag -Werror"
   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([])],
     [check_cflag=yes],
     [check_cflag=no])
@@ -84,9 +85,10 @@
 ])
 
 AC_DEFUN([CHECK_CXXFLAG], [
+  cxxflag=$(echo "$1" | sed -e 's,^-Wno-,-W,')
   AC_MSG_CHECKING([if g++ supports $1])
   old_CXXFLAGS=$CXXFLAGS
-  CXXFLAGS="$CXXFLAGS $1 -Werror"
+  CXXFLAGS="$CXXFLAGS $cxxflag -Werror"
   AC_LANG_PUSH([C++])
   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([])],
     [check_cxxflag=yes],
