--- a/Makefile	2013-02-20 16:05:12.000000000 +0100
+++ b/Makefile	2013-02-26 10:01:10.202295020 +0100
@@ -1,7 +1,8 @@
 FLAGS0=-fPIE -Wall -pedantic -Wno-long-long -fvisibility=hidden -fstack-protector --param ssp-buffer-size=4
 GLIB=`pkg-config --cflags glib-2.0`
-CCFLAGS0=-c -m64 -fPIC -D_GNU_SOURCE=1 -I. $(GLIB)
-CXXFLAGS0=-m64 -Wno-variadic-macros $(GLIB)
+ZMQ=`pkg-config --cflags libzmq`
+CCFLAGS0=-c -m64 -fPIC -D_GNU_SOURCE=1 -I. $(GLIB) $(ZMQ)
+CXXFLAGS0=-m64 -Wno-variadic-macros $(GLIB) $(ZMQ)
 LIBS=-lzmq -lglib-2.0
 TESTLIBS=-Llib/gtest -lgtest $(LIBS)
 
