diff --git a/mrn_mysql_compat.h b/mrn_mysql_compat.h
index d2b227ff..337de968 100644
--- a/mrn_mysql_compat.h
+++ b/mrn_mysql_compat.h
@@ -362,6 +362,7 @@
 
 #if defined(MRN_MARIADB_P) &&                                           \
   ((MYSQL_VERSION_ID >= 100207) ||                                      \
+   ((MYSQL_VERSION_ID >= 100126) && (MYSQL_VERSION_ID < 100200)) ||     \
    ((MYSQL_VERSION_ID >= 50557) && (MYSQL_VERSION_ID < 100000)))
 #  define mrn_create_partition_name(out,                                \
                                     out_length,                         \
