diff -ru mariadb-10.4.34.orig/mysql-test/main/ssl_crl.result mariadb-10.4.34/mysql-test/main/ssl_crl.result
--- mariadb-10.4.34.orig/mysql-test/main/ssl_crl.result	2024-05-06 17:18:43.000000000 +0200
+++ mariadb-10.4.34/mysql-test/main/ssl_crl.result	2026-01-06 11:43:21.280984670 +0100
@@ -2,4 +2,4 @@
 Variable_name	Value
 Ssl_version	TLS_VERSION
 # try logging in with a certificate in the server's --ssl-crl : should fail
-ERROR 2026 (HY000): TLS/SSL error: sslv3 alert certificate revoked
+ERROR 2026 (HY000): TLS/SSL error: ssl/tls alert certificate revoked
diff -ru mariadb-10.4.34.orig/mysql-test/main/ssl_crl.test mariadb-10.4.34/mysql-test/main/ssl_crl.test
--- mariadb-10.4.34.orig/mysql-test/main/ssl_crl.test	2024-05-06 17:18:43.000000000 +0200
+++ mariadb-10.4.34/mysql-test/main/ssl_crl.test	2026-01-06 12:56:54.513034763 +0100
@@ -8,6 +8,6 @@
 
 --echo # try logging in with a certificate in the server's --ssl-crl : should fail
 # OpenSSL 1.1.1a correctly rejects the certificate, but the error message is different
---replace_regex /ERROR 2013 \(HY000\): Lost connection to MySQL server at '.*', system error: [0-9]+/ERROR 2026 (HY000): TLS\/SSL error: sslv3 alert certificate revoked/
+--replace_regex /ERROR 2013 \(HY000\): Lost connection to MySQL server at '.*', system error: [0-9]+/ERROR 2026 (HY000): TLS\/SSL error: ssl\/tls alert certificate revoked/
 --error 1
 --exec $MYSQL --ssl-ca=$MYSQL_TEST_DIR/std_data/cacert.pem --ssl-key=$MYSQL_TEST_DIR/std_data/client-key.pem --ssl-cert=$MYSQL_TEST_DIR/std_data/client-cert.pem test -e "SHOW STATUS LIKE 'Ssl_version'" 2>&1
