diff --git a/extra/mariabackup/backup_copy.cc b/extra/mariabackup/backup_copy.cc
index e8f2e7c7d56..3829a1279f9 100644
--- a/extra/mariabackup/backup_copy.cc
+++ b/extra/mariabackup/backup_copy.cc
@@ -134,6 +134,9 @@ static bool backup_files_from_datadir(ds_ctxt_t *ds_data,
                                       const char *dir_path,
                                       const char *prefix);
 
+static bool backup_mroonga_files_from_datadir(ds_ctxt_t *ds_data,
+                                              const char *dir_path);
+
 /************************************************************************
 Retirn true if character if file separator */
 bool
@@ -1505,7 +1508,8 @@ bool backup_start(ds_ctxt *ds_data, ds_ctxt *ds_meta,
 	                               "aws-kms-key") ||
             !backup_files_from_datadir(ds_data,
                                        aria_log_dir_path,
-                                       "aria_log")) {
+                                       "aria_log") ||
+            !backup_mroonga_files_from_datadir(ds_data, fil_path_to_mysql_datadir)) {
 		return false;
 	}
 
@@ -1725,7 +1729,9 @@ ibx_copy_incremental_over_full()
 						      "aws-kms-key")) ||
 		    !(ret = backup_files_from_datadir(ds_data,
 						      xtrabackup_incremental_dir,
-						      "aria_log")))
+						      "aria_log")) ||
+		    !(ret = backup_mroonga_files_from_datadir(ds_data,
+						      xtrabackup_incremental_dir)))
 			goto cleanup;
 
 		/* copy supplementary files */
@@ -2307,6 +2313,39 @@ static bool backup_files_from_datadir(ds_ctxt_t *ds_data,
 	return ret;
 }
 
+static bool backup_mroonga_files_from_datadir(ds_ctxt_t *ds_data,
+                                              const char *dir_path)
+{
+	os_file_dir_t dir = os_file_opendir(dir_path);
+	if (dir == IF_WIN(INVALID_HANDLE_VALUE, nullptr)) return false;
+
+	os_file_stat_t info;
+	bool ret = true;
+	while (os_file_readdir_next_file(dir_path, dir, &info) == 0) {
+
+		if (info.type != OS_FILE_TYPE_FILE)
+			continue;
+
+		const char *pname = strrchr(info.name, OS_PATH_SEPARATOR);
+		if (!pname)
+			pname = info.name;
+
+		if (!strstr(pname, ".mrn"))
+			continue;
+
+		if (xtrabackup_prepare && xtrabackup_incremental_dir &&
+			file_exists(info.name))
+			unlink(info.name);
+
+		std::string full_path(dir_path);
+		full_path.append(1, OS_PATH_SEPARATOR).append(info.name);
+		if (!(ret = ds_data->copy_file(full_path.c_str() , info.name, 1)))
+			break;
+	}
+	os_file_closedir(dir);
+	return ret;
+}
+
 
 static int rocksdb_remove_checkpoint_directory()
 {
