commit 4a40e81d17d1df28cfeca56d3d9c8f26d0132cfa
Author: Kouhei Sutou <kou@clear-code.com>
Date:   Sun Oct 18 17:23:28 2015 +0900

    Use public API
    
    It fixes a build error on MariaDB 10.1.8.

diff --git a/ha_mroonga.cpp b/ha_mroonga.cpp
index 819de30..bfd9631 100644
--- a/ha_mroonga.cpp
+++ b/ha_mroonga.cpp
@@ -9620,13 +9620,13 @@ void ha_mroonga::check_count_skip(key_part_map start_key_part_map,
 
         if (where->type() == Item::FUNC_ITEM) {
           Item_func *func_item = static_cast<Item_func *>(where);
-          if (func_item->arg_count == 0) {
+          if (func_item->argument_count() == 0) {
             break;
           }
           target = func_item->key_item();
           where = where->next;
           if (func_item->arguments()[0] == where) {
-            uint n_args = func_item->arg_count;
+            uint n_args = func_item->argument_count();
             for (; n_args > 0; --n_args) {
               where = where->next;
             }
