--- ha_mroonga.cc.orig	2012-04-11 17:59:24.171323133 +0200
+++ ha_mroonga.cc	2012-04-11 18:01:50.431325458 +0200
@@ -10319,10 +10319,11 @@
 bool ha_mroonga::wrapper_auto_repair() const
 {
   bool res;
+  int ha_err = 0;
   MRN_DBUG_ENTER_METHOD();
   MRN_SET_WRAP_SHARE_KEY(share, table->s);
   MRN_SET_WRAP_TABLE_KEY(this, table);
-  res = wrap_handler->auto_repair();
+  res = wrap_handler->auto_repair(ha_err);
   MRN_SET_BASE_SHARE_KEY(share, table->s);
   MRN_SET_BASE_TABLE_KEY(this, table);
   DBUG_RETURN(res);
@@ -10330,9 +10331,10 @@
 
 bool ha_mroonga::storage_auto_repair() const
 {
+  int ha_err = 0;
   MRN_DBUG_ENTER_METHOD();
   // XXX: success is valid variable name?
-  bool success = handler::auto_repair();
+  bool success = handler::auto_repair(ha_err);
   DBUG_RETURN(success);
 }
 
