--- Zope2-2.13.30/src/Shared/DC/Scripts/Signature.py	2022-04-25 08:05:09.312966168 +0000
+++ Zope2-2.13.30/src/Shared/DC/Scripts/Signature.py	2022-04-25 08:06:20.120743425 +0000
@@ -35,7 +35,7 @@ def _setFuncSignature(self, defaults=None, varnames=(), argcount=-1):
         argcount = len(varnames)
     # Generate a change only if we have to.
     if self.func_defaults != defaults:
-        self.func_defaults = defaults
+        self.func_defaults = self.__defaults__ = defaults
     code = FuncCode(varnames, argcount)
     if self.func_code != code:
-        self.func_code = code
+        self.func_code = self.__code__ = code
