diff -u old/Makefile new/Makefile
--- old/Makefile
+++ new/Makefile
@@ -280,6 +280,7 @@
 SYSTEM=unix
 TOOLS=tools
 TOUCH=touch
+INSTALL=install
 
 
 # Primary build command
@@ -580,7 +581,6 @@
 	@echo +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 	@echo
 	@echo Do you want to continue this build anyway?  Type y or n please:
-	@$(SH) -c 'read x; case "$$x" in y) exit 0;; *) (make nounenc;exit 1);; esac'
 
 nounenc:
 	@echo +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
@@ -639,7 +639,6 @@
 	@echo +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 	@echo
 	@echo Do you want to build with IPv6 anyway?  Type y or n please:
-	@$(SH) -c 'read x; case "$$x" in y) exit 0;; *) (make noip6;exit 1);; esac'
 	@echo OK, I will remember that you really want to build with IPv6.
 	@echo You will not see this message again.
 	@$(TOUCH) ip6
@@ -731,6 +730,15 @@
 	$(SH) -c '$(RM) an ua OSTYPE SPECIALS c-client mtest imapd ipopd mailutil mlock dmail tmail || true'
 	$(CD) tools;$(MAKE) clean
 
+install:
+	$(INSTALL) -v -d $(PREFIX)/bin $(PREFIX)/include $(PREFIX)/lib
+	$(INSTALL) -v c-client/*.h $(PREFIX)/include
+	$(INSTALL) -v c-client/c-client.a $(PREFIX)/lib
+	for x in ipopd mailutil imapd dmail mlock mtest tmail; \
+	do for x in $$x/*; do [ ! -x $$x ] || echo $$x; done; \
+	done |xargs $(INSTALL) -v -t $(PREFIX)/bin
+	$(INSTALL) -v -t $(PREFIX) ip6 OSTYPE SPECIALS
+
 
 # A monument to a hack of long ago and far away...
 love:
