--- config	2012-05-19 18:03:04.000000000 +0200
+++ config2	2012-05-19 18:03:37.000000000 +0200
@@ -553,15 +553,7 @@
 	fi ;;
   i?86-apple-darwin*)
 	ISA64=`(sysctl -n hw.optional.x86_64) 2>/dev/null`
-	if [ "$ISA64" = "1" -a -z "$KERNEL_BITS" ]; then
-	    echo "WARNING! If you wish to build 64-bit library, then you have to"
-	    echo "         invoke './Configure darwin64-x86_64-cc' *manually*."
-	    if [ "$TEST" = "false" -a -t 1 ]; then
-	      echo "         You have about 5 seconds to press Ctrl-C to abort."
-	      (trap "stty `stty -g`" 2 0; stty -icanon min 0 time 50; read waste) <&1
-	    fi
-	fi
-	if [ "$ISA64" = "1" -a "$KERNEL_BITS" = "64" ]; then
+	if [ "$ISA64" = "1" ]; then
 	    OUT="darwin64-x86_64-cc"
 	else
 	    OUT="darwin-i386-cc"
