diff -ur mysqlclient-1.3.10.orig/_mysql.c mysqlclient-1.3.10/_mysql.c
--- mysqlclient-1.3.10.orig/_mysql.c	2017-01-04 13:47:08.000000000 +0100
+++ mysqlclient-1.3.10/_mysql.c	2017-07-18 01:28:29.021729572 +0200
@@ -1060,7 +1060,7 @@
 	if (self && PyModule_Check((PyObject*)self))
 		self = NULL;
 	if (self && self->open) {
-#if MYSQL_VERSION_ID >= 50707 && !defined(MARIADB_BASE_VERSION)
+#if MYSQL_VERSION_ID >= 50707 && !defined(MARIADB_BASE_VERSION) && !defined(MARIADB_VERSION_ID)
 		len = mysql_real_escape_string_quote(&(self->connection), out, in, size, '\'');
 #else
 		len = mysql_real_escape_string(&(self->connection), out, in, size);
@@ -1118,7 +1118,7 @@
 	out = PyBytes_AS_STRING(str);
 	check_server_init(NULL);
 	if (self && self->open) {
-#if MYSQL_VERSION_ID >= 50707 && !defined(MARIADB_BASE_VERSION)
+#if MYSQL_VERSION_ID >= 50707 && !defined(MARIADB_BASE_VERSION) && !defined(MARIADB_VERSION_ID)
 		len = mysql_real_escape_string_quote(&(self->connection), out+1, in, size, '\'');
 #else
 		len = mysql_real_escape_string(&(self->connection), out+1, in, size);
