--- perl-5.20.0/Configure.orig	2014-05-28 15:27:22.865060361 +0200
+++ perl-5.20.0/Configure	2014-05-28 15:32:14.297064994 +0200
@@ -8277,16 +8277,7 @@
 	case "$dflt" in
 	    none) dflt='' ;;
 	esac
-	for thisflag in $ldflags; do
-	    case "$thisflag" in
-		-L*|-R*|-Wl,-R*)
-		    case " $dflt " in
-			*" $thisflag "*) ;;
-			*) dflt="$dflt $thisflag" ;;
-		    esac
-		    ;;
-	    esac
-	done
+	dflt="$dflt $ldflags"
 
 	case "$dflt" in
 	    ''|' ') dflt='none' ;;
