--- configure.orig	2013-07-05 15:37:28.000000000 +0200
+++ configure	2013-07-05 15:47:48.000000000 +0200
@@ -4680,52 +4680,6 @@
 if test "x$ABI" != xstandard ; then
   { $as_echo "$as_me:${as_lineno-$LINENO}: Compiler uses $ABI-bit ABI. To change, set CC." >&5
 $as_echo "$as_me: Compiler uses $ABI-bit ABI. To change, set CC." >&6;}
-  if test "$libdir" = '${exec_prefix}/lib' ; then
-    # Try setting a better default
-    case "$host_cpu:$host_os:$ABI" in
-      *:solaris*:32|*:sunos*:32)
-	libdir='${exec_prefix}/lib'
-	;;
-      *:solaris*:64|*:sunos*:64)
-	libdir='${exec_prefix}/lib/64'
-	;;
-      # Linux conventions are a mess... According to the Linux File
-      # Hierarchy Standard, all architectures except IA64 puts 32-bit
-      # libraries in lib, and 64-bit in lib64. Some distributions,
-      # e.g., Fedora and Gentoo, adhere to this standard, while at
-      # least Debian has decided to put 64-bit libraries in lib and
-      # 32-bit libraries in lib32.
-
-      # We try to figure out the convention, except if we're cross
-      # compiling. We use lib${ABI} if /usr/lib${ABI} exists and
-      # appears to not be a symlink to a different name.
-      *:linux*:32|*:linux*:64)
-	if test "$cross_compiling" = yes ; then
-          { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Cross compiling for linux. Can't guess if libraries go in lib${ABI} or lib." >&5
-$as_echo "$as_me: WARNING: Cross compiling for linux. Can't guess if libraries go in lib${ABI} or lib." >&2;}; 	else
-	  # The dash builtin pwd tries to be "helpful" and remember
-	  # symlink names. Use -P option, and hope it's portable enough.
-	  test -d /usr/lib${ABI} \
-	    && (cd /usr/lib${ABI} && pwd -P | grep >/dev/null "/lib${ABI}"'$') \
-	    && libdir='${exec_prefix}/'"lib${ABI}"
-	fi
-	;;
-      # On freebsd, it seems 32-bit libraries are in lib32,
-      # and 64-bit in lib. Don't know about "kfreebsd", does
-      # it follow the Linux fhs conventions?
-      *:freebsd*:32)
-	libdir='${exec_prefix}/lib32'
-	;;
-      *:freebsd*:64)
-	libdir='${exec_prefix}/lib'
-	;;
-      *)
-        { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Don't know where to install $ABI-bit libraries on this system." >&5
-$as_echo "$as_me: WARNING: Don't know where to install $ABI-bit libraries on this system." >&2;};
-    esac
-    { $as_echo "$as_me:${as_lineno-$LINENO}: Libraries to be installed in $libdir." >&5
-$as_echo "$as_me: Libraries to be installed in $libdir." >&6;}
-  fi
 fi
 
 # Select assembler code
