diff -ur dmtx-utils-0.7.4.orig/configure.ac dmtx-utils-0.7.4/configure.ac
--- dmtx-utils-0.7.4.orig/configure.ac	2011-06-03 07:13:10.000000000 +0200
+++ dmtx-utils-0.7.4/configure.ac	2016-05-06 22:56:23.129582472 +0200
@@ -60,7 +60,7 @@
 fi
 
 if test x$dmtxread = xyes -o x$dmtxwrite = xyes; then
-   PKG_CHECK_MODULES(MAGICK, Wand >= 6.2.4, [], AC_MSG_ERROR([dmtxread/dmtxwrite requires Wand >= 6.2.4]))
+   PKG_CHECK_MODULES(MAGICK, MagickWand >= 6.2.4, [], AC_MSG_ERROR([dmtxread/dmtxwrite requires MagickWand >= 6.2.4]))
    AC_SUBST(MAGICK_CFLAGS)
    AC_SUBST(MAGICK_LIBS)
 fi
diff -ur dmtx-utils-0.7.4.orig/dmtxread/dmtxread.c dmtx-utils-0.7.4/dmtxread/dmtxread.c
--- dmtx-utils-0.7.4.orig/dmtxread/dmtxread.c	2011-06-03 07:13:10.000000000 +0200
+++ dmtx-utils-0.7.4/dmtxread/dmtxread.c	2016-05-06 23:17:08.190667778 +0200
@@ -114,7 +114,7 @@
          }
 
          /* Copy pixels to known format */
-         success = MagickGetImagePixels(wand, 0, 0, width, height, "RGB", CharPixel, pxl);
+         success = MagickExportImagePixels(wand, 0, 0, width, height, "RGB", CharPixel, pxl);
          if(success == MagickFalse || pxl == NULL) {
             CleanupMagick(&wand, DmtxTrue);
             FatalError(EX_OSERR, "malloc() error");
diff -ur dmtx-utils-0.7.4.orig/dmtxread/dmtxread.h dmtx-utils-0.7.4/dmtxread/dmtxread.h
--- dmtx-utils-0.7.4.orig/dmtxread/dmtxread.h	2011-06-03 07:13:10.000000000 +0200
+++ dmtx-utils-0.7.4/dmtxread/dmtxread.h	2016-05-06 23:01:38.871722717 +0200
@@ -33,7 +33,7 @@
 #include <math.h>
 #include <stdarg.h>
 #include <assert.h>
-#include <wand/magick-wand.h>
+#include <MagickWand/MagickWand.h>
 #include <dmtx.h>
 #include "../common/dmtxutil.h"
 
diff -ur dmtx-utils-0.7.4.orig/dmtxwrite/dmtxwrite.h dmtx-utils-0.7.4/dmtxwrite/dmtxwrite.h
--- dmtx-utils-0.7.4.orig/dmtxwrite/dmtxwrite.h	2011-06-03 07:13:10.000000000 +0200
+++ dmtx-utils-0.7.4/dmtxwrite/dmtxwrite.h	2016-05-06 23:01:38.871722717 +0200
@@ -34,7 +34,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <assert.h>
-#include <wand/magick-wand.h>
+#include <MagickWand/MagickWand.h>
 #include <dmtx.h>
 #include "../common/dmtxutil.h"
 
