--- poptconfig.c~	2011-09-10 22:49:30.802250000 +0800
+++ poptconfig.c	2011-09-10 23:46:30.572048000 +0800
@@ -45,8 +45,9 @@ extern int glob_pattern_p (const char *_
 #if !defined(__GLIBC__)
 /* Return nonzero if PATTERN contains any metacharacters.
    Metacharacters can be quoted with backslashes if QUOTE is nonzero.  */
+#define glob_pattern_p glob_pattern_p_s
 static int
-glob_pattern_p (const char * pattern, int quote)
+glob_pattern_p_s(const char * pattern, int quote)
 	/*@*/
 {
     const char * p;
