diff --git a/miscutils/ubi_tools.c b/miscutils/ubi_tools.c
index 2a426db..35f44ac 100644
--- a/miscutils/ubi_tools.c
+++ b/miscutils/ubi_tools.c
@@ -60,6 +60,16 @@
 //kbuild:lib-$(CONFIG_UBIUPDATEVOL) += ubi_tools.o
 
 #include "libbb.h"
+#if 1
+// __packed hack for broken kernel headers ubi-user.h
+// See: https://lkml.org/lkml/2011/6/22/25
+//   Once ubi-user.h is correctly installed without "__packed", or kernel
+//   guys correct this another way (with a compiler header??), this can
+//   go away.
+# ifndef __packed
+#  define __packed __attribute__((packed))
+# endif // ndef __packed
+#endif // 1
 #include <mtd/ubi-user.h>
 
 #define OPTION_M  (1 << 0)
