--- jdk-6u27-linux-x64.bin.orig	2011-09-27 11:02:14.000000000 +0200
+++ jdk-6u27-linux-x64.bin	2011-09-27 10:38:01.000000000 +0200
@@ -81,7 +81,7 @@
 trap 'rm -f $outname; exit 1' HUP INT QUIT TERM
 echo "Unpacking..."
 tail ${tail_args} +189 "$0" > $outname
-if [ -x /usr/bin/sum ]; then
+if [ -x /usr/bin/null ]; then
     echo "Checksumming..."
 
     sum=`/usr/bin/sum $outname`
@@ -169,7 +169,7 @@
     fi
 
     # Service Tag support and JDK product registration
-    register_JDK "$javahome" "${BINARY_NAME}" "$1"
+    # register_JDK "$javahome" "${BINARY_NAME}" "$1"
 
 else
     if [ "$1" = "-x" ]; then
