diff -ur trafficserver-9.1.1.orig/src/traffic_server/InkAPI.cc trafficserver-9.1.1/src/traffic_server/InkAPI.cc
--- trafficserver-9.1.1.orig/src/traffic_server/InkAPI.cc	2021-10-29 15:38:50.000000000 +0000
+++ trafficserver-9.1.1/src/traffic_server/InkAPI.cc	2022-01-31 10:37:24.675347079 +0000
@@ -5443,7 +5443,11 @@
     break;
   case HttpTransact::CACHE_LOOKUP_HIT_WARNING:
   case HttpTransact::CACHE_LOOKUP_HIT_FRESH:
-    *lookup_status = TS_CACHE_LOOKUP_HIT_FRESH;
+    if (HttpTransact::need_to_revalidate(&sm->t_state)) {
+      *lookup_status = TS_CACHE_LOOKUP_MISS;
+    } else {
+      *lookup_status = TS_CACHE_LOOKUP_HIT_FRESH;
+    }
     break;
   case HttpTransact::CACHE_LOOKUP_SKIPPED:
     *lookup_status = TS_CACHE_LOOKUP_SKIPPED;
